project/include/ui/FileDialog.h

```diff
@@ -13,10 +13,23 @@ namespace lime {
 
 		public:
 
-			static std::wstring* OpenDirectory (std::wstring* title = 0, std::wstring* filter = 0, std::wstring* defaultPath = 0);
-			static std::wstring* OpenFile (std::wstring* title = 0, std::wstring* filter = 0, std::wstring* defaultPath = 0);
-			static void OpenFiles (std::vector<std::wstring*>* files, std::wstring* title = 0, std::wstring* filter = 0, std::wstring* defaultPath = 0);
-			static std::wstring* SaveFile (std::wstring* title = 0, std::wstring* filter = 0, std::wstring* defaultPath = 0);
+		#ifdef HX_WINDOWS
+			// Path separator used by OpenFiles
+			static const wchar_t PATH_SEPARATOR = L'|';
+
+			static const wchar_t* OpenDirectory (const wchar_t* title = 0, const wchar_t* filter = 0, const wchar_t* defaultPath = 0);
+			static const wchar_t* OpenFile (const wchar_t* title = 0, const wchar_t* filter = 0, const wchar_t* defaultPath = 0);
+			static const wchar_t* OpenFiles (const wchar_t* title = 0, const wchar_t* filter = 0, const wchar_t* defaultPath = 0);
+			static const wchar_t* SaveFile (const wchar_t* title = 0, const wchar_t* filter = 0, const wchar_t* defaultPath = 0);
+		#else
+			// Path separator used by OpenFiles
+			static const char PATH_SEPARATOR = '|';
+
+			static const char* OpenDirectory (const char* title = 0, const char* filter = 0, const char* defaultPath = 0);
+			static const char* OpenFile (const char* title = 0, const char* filter = 0, const char* defaultPath = 0);
+			static const char* OpenFiles (const char* title = 0, const char* filter = 0, const char* defaultPath = 0);
+			static const char* SaveFile (const char* title = 0, const char* filter = 0, const char* defaultPath = 0);
+		#endif
```

Comment:
On the one hand, I'm pretty sure this is what typedefs are for. It would be really nice not to have this kind of duplicate code.

On the other hand, if we aren't careful, new typedefs would just mean new instances where conditional compilation is needed. For instance, a type that mapped to `wchar_t` on Windows and `char` elsewhere doesn't help when Windows code needs to use `char`, so we'd still need `#ifdef HX_WINDOWS`.